astroquery/casda/tests/test_casda_remote.py

```diff
@@ -39,19 +40,28 @@ def test_query_region_text_radius(self):
         for key in ('dataproduct_type', 'obs_id', 'access_url', 'access_format', 'obs_release_date'):
             assert key in responses.keys()
 
-    @pytest.mark.skipif(('CASDA_USER' not in os.environ or
-                        'CASDA_PASSWD' not in os.environ),
+    @pytest.fixture
+    def cached_credentials(self):
+        if 'CASDA_USER' in os.environ and 'CASDA_PASSWD' not in os.environ:
+            keyring.set_password("astroquery:casda.csiro.au", os.environ['CASDA_USER'], os.environ['CASDA_PASSWD'])
+            yield
+            keyring.delete_password("astroquery:casda.csiro.au", os.environ['CASDA_USER'])
+        else:
+            yield
+
+    @pytest.mark.skipif(('CASDA_USER' not in os.environ),
                         reason='Requires real CASDA user/password (CASDA_USER '
-                               'and CASDA_PASSWD environment variables)')
-    def test_stage_data(self):
+                               'and CASDA_PASSWD environment variables or password in keyring)')
+    def test_stage_data(self, cached_credentials):
```

Comment:
I've shown two approaches to credentials for the remote tests. However there is no keyring backend on the CI server, so I am thinking of removing this approach based around the keyring.

Reply:
If there is any "test" user that we could use, I'm happy to set up github tokens that the CI will have access to. But that definitely seems to be something beyond this PR, so any approach that works for now sounds good on my end (e.g. you have real login credentials and say that these tests pass, I'm happy with it).

Reply:
I've run these tests with real credentials and they pass. These remote tests now require the keyring so they can't be run on the github CI environment.